app/assets/scripts/map.js

```diff
@@ -0,0 +1,121 @@
+// Global Marker Wrappers and Map
+var globalMarkerWrappers = null;
+var globalMap = null;
+
+// Initial map center coordinates
+INITIAL_CENTER_LAT = 39.952;
+INITIAL_CENTER_LONG = -75.195;
+initial_coords = new google.maps.LatLng(INITIAL_CENTER_LAT, INITIAL_CENTER_LONG);
+
+// Bounds for the Date Slider. End date is current date.
+var BOUNDS_MIN;
+var BOUNDS_MAX = new Date();
+
+// Make a type for an object that encapsulates a GoogleMaps marker
+// object, the corresponding date, and the corresponding message to
+// display for the marker.
+function MarkerWrapper(actualMarker, incidentDate, contentString) {
+    this.actualMarker = actualMarker;
+    this.incidentDate = incidentDate;
+    this.contentString = contentString;
+}
+
+// Take a list of MarkerWrappers and a map, put them on the map, set the
+// minimum date, and set the location bounds
+function storeMarkerState(markerWrappers, map, minDate, bounds) {
+    globalMarkerWrappers = markerWrappers;
+    globalMap = map;
+    for (mw = 0; mw < globalMarkerWrappers.length; mw++)
+    {
+        (globalMarkerWrappers[mw].actualMarker).setMap(globalMap);
+    }
+    BOUNDS_MIN = minDate;
+    map.fitBounds(bounds);
+}
+
+// Use Google geocoder to update geolocation given an address through
+// the address search box
+function update_center() {
+    geocoder = new google.maps.Geocoder();
+    address = $("#address").val();
+    if (geocoder) {
+        geocoder.geocode( { 'address': address},
+        function(results, status) {
+            if (status == google.maps.GeocoderStatus.ZERO_RESULTS) {
+                $('#address_error').modal('show');
+            }
+            else if (status != google.maps.GeocoderStatus.OK) {
+                $('#geocode_error').modal('show');
+            }
+            else {
+                globalMap.setCenter(results[0].geometry.location);
+            }
+            $('.close.icon').on('click', function() {
+                $(this).parent().hide();
+            });
+        });
+    }
+}
+
+$('.close.icon').on('click', function() {
+    $(this).parent().hide();
+});
+
+// Get address submit event
+$(document).ready(function() {
```

Comment:
I think that because there are references to functions in the code outside the file, this would cause problems related to scope (I got errors when I just tried it).